lib/proxy/log_buffer/decorator.rb

```diff
@@ -0,0 +1,70 @@
+require 'proxy/log_buffer/decorator'
+require 'proxy/log_buffer/buffer'
+
+module Proxy::LogBuffer
+  class Decorator
+    def initialize(logger)
+      @logger = logger
+    end
+
+    def add(severity, message = nil, progname = nil, backtrace = nil, a_module = 'core', &block)
```

Comment:
nitpick: &block is not needed. you can still pass blocks
@dLobatog told me that passing &block makes it slower, but i have no data to back this up :)